src/datasets/utils/metadata.py

```diff
@@ -173,14 +194,42 @@ def to_builder_configs(self, builder_config_cls):
                     You can resolve data files with either .resolve_data_files_locally() or
                     .resolve_data_files_in_dataset_repository() method."""
                 )
+        ignored_params = [
+            param
+            for meta_config in metadata_configs.values()
+            for param in meta_config
+            if hasattr(builder_config_cls, param) and param != "default"
+        ]
+        if ignored_params:
+            logger.warning(
+                f"Some datasets params were ignored: {ignored_params}. "
+                "Make sure to use only valid params for the dataset builder and to have "
+                "a up-to-date version of the `datasets` library."
+            )
         return [
             builder_config_cls(
                 name=name,
-                **{param: value for param, value in meta_config.items() if hasattr(builder_config_cls, param)},
+                **{
+                    param: value
+                    for param, value in meta_config.items()
+                    if hasattr(builder_config_cls, param) and param != "default"
```

Comment:
what does it mean that config parameter is named "default"? how that might be possible? Do you mean we can now write smth like 
```yaml
builder_config:
  - config_name: v1
    default: true
    ...
```
to make a custom config default?

Reply:
Yes exactly - forgot to mention it in the OP